corehq/apps/app_manager/helpers/validators.py

```diff
@@ -950,34 +948,21 @@ def validate_for_module(self, module):
                                 errors.append(dict(type='bad form link', **meta))
                         elif linked_module.unique_id != linked_form.get_module().unique_id:
                             errors.append(dict(type='bad form link', **meta))
-                    else:
-                        linked_module = linked_form.get_module()
-
-                else:
-                    try:
-                        linked_module = self.app.get_module_by_unique_id(form_link.module_unique_id)
-                    except ModuleNotFoundException:
-                        errors.append(dict(type='bad form link', **meta))
-                if linked_module:
-                    if linked_module.is_multi_select():
-                        errors.append(dict(type="multi select form links", **meta))
-                    if linked_module.root_module and linked_module.root_module.is_multi_select():
-                        errors.append(dict(type='parent multi select form links', **meta))
+                        if not linked_module.is_multi_select() and self.form.get_module().is_multi_select():
+                            errors.append(dict(type='multi select form links', **meta))
         elif self.form.post_form_workflow == WORKFLOW_MODULE:
             if module.put_in_root:
                 errors.append(dict(type='form link to display only forms', **meta))
-            if module.root_module and module.root_module.is_multi_select():
-                errors.append(dict(type='parent multi select form links', **meta))
         elif self.form.post_form_workflow == WORKFLOW_PARENT_MODULE:
             if not module.root_module:
                 errors.append(dict(type='form link to missing root', **meta))
             elif module.root_module.put_in_root:
                 errors.append(dict(type='form link to display only forms', **meta))
-            elif module.root_module.is_multi_select():
-                errors.append(dict(type='parent multi select form links', **meta))
         elif self.form.post_form_workflow == WORKFLOW_PREVIOUS:
-            if module.is_multi_select() or module.root_module and module.root_module.is_multi_select():
-                errors.append(dict(type='previous multi select form links', **meta))
+            if module.root_module:
+                if (module.is_multi_select() and not module.root_module.is_multi_select()) or \
+                        (not module.is_multi_select() and module.root_module.is_multi_select()):
```

Comment:
nitpick: I think it would be cleaner like `result = (module.is_multi_select() != module.root_module.is_multi_select())` then `if result:` with a contextual `result` variable name. Can also have it with `module.root_module` if statement like `if module.root_module and result`:

Reply:
Another possibility I considered but thought it was less intutive when reading it was to replace the check in line 963 & 964 with `if not((module.is_multi_select() == module.root_module.is_multi_select())`